data/binding/bindlists.go

```diff
@@ -0,0 +1,404 @@
+// auto-generated
+// **** THIS FILE IS AUTO-GENERATED, PLEASE DO NOT EDIT IT **** //
+
+package binding
+
+// BoolList supports binding a list of bool values.
+//
+// Since: 2.0.0
+type BoolList interface {
+	DataList
+
+	Append(bool)
+	Get(int) bool
+	Prepend(bool)
+	Set(int, bool)
+}
+
+// NewBoolList returns a bindable list of bool values.
+//
+// Since: 2.0.0
+func NewBoolList() BoolList {
+	return &boundBoolList{}
+}
+
+// BindBoolList returns a bound list of bool values, based on the contents of the passed slice.
+//
+// Since: 2.0.0
+func BindBoolList(v *[]bool) BoolList {
+	if v == nil {
+		return NewBoolList()
+	}
+
+	b := &boundBoolList{val: v}
+
+	for _, i := range *v {
+		b.Append(i)
+	}
```

Comment:
Not sure about this, wouldn't calling `BindBoolList(&[]bool{true})` initialize `boundBoolList` with the slice pointer, and then this `for` would append the data again, resulting in `&[]bool{true, true}`?

Reply:
Oh darn, this was inintially split into value and binding init. 
Let me go see

Reply:
Fixed, good catch.
The tests missed it because the binding count was correct, but our data was duplicated

Reply:
Not sure about the fix, is `val` ever used now?

Reply:
The get looks it up as an optimisation.
Additionally we need to retain a reference to determine what changes when we get into reload requests - whatever api approach we take.

Reply:
> The get looks it up as an optimisation.

It will always be nil tho

Reply:
Oh I see. good catch.
Better fix in place - what I should have done first time.
Also tests to validate this time!